lesson_1.py

```diff
@@ -1,58 +1,87 @@
-"""
-Challenge:
-
-#. Implement ``Product`` class
-#. Each ``Product`` instance should implement properties:
-    * ``name`` - a product's name, like apple, cheese etc.
-    * ``price`` - a price for a single unit
-#. ``Product`` instance should have a behavior of calculating total
-   price for a specified quantity of goods
-#. Implement ``ShoppingCart`` class
-#. ``ShoppingCart`` instance should combine products instances and
-    corresponding purchased quantities.
-#. ``ShoppingCart`` instance should implement a method to calculate
-   the total price of entire cart.
-
-"""
-import random
 from typing import Union
 
 
 class Product():
-    def __init__(self, name: str, price: int):
+    def __init__(self, name: str, price: float):
         self.name = name
         self.price = price
 
     def calculate_cost(self, quantity: Union[int, float]):
         return round(quantity * self.price, 2)
 
+    def __str__(self):
+        return self.name
+
+    def __float__(self):
+        return self.price
+
 
 class Shopping_Cart():
     def __init__(self):
         self.products: List[Product] = []
         self.quantities: List[Union[int, float]] = []
 
     def add_to_cart(self, product: Product, quantity: Union[int, float]):
-        self.products.append(product)
-        self.quantities.append(quantity)
+        if product not in self.products:
+            self.products.append(product)
+            self.quantities.append(quantity)
+        else:
+            self.quantities[self.products.index(product)] += quantity
```

Comment:
Не помилка, але от так красивіше та ліпше читати
```suggestion
            idx = self.products.index(product)
            self.quantities[idx] += quantity
```